### PR TITLE
[jit] clang format jit

### PR DESCRIPTION
**Pull request by suo**
The PR clang-formats everything in `torch/csrc/jit/` and adds it to the pre-commit hook. 

Here is a list of non-mechanical changes:
- I went over each file and fixed up whenever I could tell that clang-format was clobbering comment formatting.
- Made the macros in register_prim_ops a little more clang-format friendly by omitting trailing commas
- Refactored autodiff.cpp to use a helper class with explicit state rather than a bunch of capturing lambdas
- Small improvements to the precommit hook clang-format

### CODE REVIEW

**Review by soumith — APPROVED** [review]

**Review by apaszke — CHANGES_REQUESTED** [review]
As far as I understand we still don't enforce clang-format on all files, nor did we even let people write some PR agains the partial check. Maybe it's better to wait for results of that before we merge a 10k line patch that doesn't do anything?

**Comment by suo** [comment]
The enforcement (such as it is) is done through the precommit hook. It's not hard enforcement, but I'm trying to get the codebase in a reasonable state before then. 

I can do things more incrementally, but it seems like that will probably be more pain for everyone involved (lots of rebasing and merge conflicts rather than ripping the bandaid off when few people have outstanding PRs).  

> nor did we even let people write some PR agains the partial check.

Do you mean that we haven't tested the hook enough?

**Comment by suo** [comment]
Spoke offline with @apaszke and decided to go ahead with band-aid ripping, followed by a warning CI check as well

**Pull request closed without merging**
